logbus/solvermon/vertexmon.go

```diff
@@ -37,111 +37,147 @@ type vertexMonitor struct {
-				internalStr, indentOp)
-			vm.fatalErrorType = logstream.FailureType_FAILURE_TYPE_OOM_KILLED
-		default:
-			errString = fmt.Sprintf(""+
-				"      The%s command\n"+
+				"      was terminated because the build system ran out of memory. If you are using a satellite or other remote buildkit, it is the remote system that ran out of memory.", internal, operation)
+	case logstream.FailureType_FAILURE_TYPE_NONZERO_EXIT:
+		return fmt.Sprintf(
+			"      The%s command\n"+
 				"          %s\n"+
-				"      did not complete successfully. Exit code %s",
-				internalStr, indentOp, exitCodeMatch)
-			vm.fatalErrorType = logstream.FailureType_FAILURE_TYPE_NONZERO_EXIT
-		}
-		vm.isFatalError = true
-	case reErrNotFound.MatchString(errString):
+				"      did not complete successfully. Exit code %d", internal, operation, exitCode)
+	case logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND:
 		m := reErrNotFound.FindStringSubmatch(errString)
-		errString = fmt.Sprintf(""+
+		return fmt.Sprintf(
 			"      The%s command\n"+
-			"          %s\n"+
-			"      failed: %s",
-			internalStr, indentOp, m[2])
-		vm.isFatalError = true
-		vm.fatalErrorType = logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND
-	case errString == "no active sessions":
-		vm.isCanceled = true
-		errString = "WARN: Canceled"
-	case strings.Contains(errString, errutil.EarthlyGitStdErrMagicString):
+				"          %s\n"+
+				"      failed: %s", internal, operation, m[2])
+	case logstream.FailureType_FAILURE_TYPE_GIT:
 		gitStdErr, shorterErr, ok := errutil.ExtractEarthlyGitStdErr(errString)
 		if ok {
-			errString = fmt.Sprintf(
-				"The%s command '%s' failed: %s\n\n%s", internalStr, vm.operation, shorterErr, gitStdErr)
-		} else {
-			errString = fmt.Sprintf(
-				"The%s command '%s' failed: %s", internalStr, vm.operation, errString)
+			return fmt.Sprintf(
+				"The%s command\n"+
+					"          %s\n"+
+					"failed: %s\n\n%s", internal, operation, shorterErr, gitStdErr)
 		}
-		vm.isFatalError = true
+		return fmt.Sprintf(
+			"The%s command\n"+
+				"          %s\n"+
+				"failed: %s", internal, operation, errString)
 	default:
-		errString = fmt.Sprintf(
-			"The%s command '%s' failed: %s", internalStr, vm.operation, errString)
+		return fmt.Sprintf(
+			"The%s command\n"+
+				"          %s\n"+
+				"failed: %s", internal, operation, errString)
 	}
+}
+
+func FormatError(operation string, errString string) string {
+	exitCode := getExitCode(errString)
+	fatalErrorType, _ := determineFatalErrorType(errString, exitCode)
+	return formatErrorMessage(errString, operation, "", fatalErrorType, exitCode)
+}
+
+func (vm *vertexMonitor) parseError() {
+	errString := vm.vertex.Error
+
+	indentOp := strings.Join(strings.Split(vm.operation, "\n"), "\n          ")
+	internalStr := ""
+	if vm.meta.Internal {
+		internalStr = " internal"
+	}
+	errString = fmt.Sprintf("%s%s", internalStr, errString)
```

Comment:
is this still needed? If yes, can we move this conversion inside the `formatErrorMessage` function?